clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessorTest.java

```diff
@@ -69,7 +70,8 @@ public void setup() {
         processor = new ApplicationEventProcessor(
             new LogContext(),
             requestManagers,
-            metadata
+            metadata,
+            mock(SubscriptionState.class)
```

Comment:
Could we add a test for the UnsubscribeEvent processing to cover what we expect on the 2 paths:
- if groupId => verify(membershipManager).leaveGroup();
- else => verify(subscriptionState).unsubscribe();

Reply:
Add related test cases. Thanks.